### ISSUE TITLE
IDE0051 false positive for private (static) method in extension block

### ISSUE THREAD

**Issue by drieseng** [dotnet/roslyn#82061]
**Version Used**: 
Visual Studio v18.2.0 / .NET SDK v10.0.102

**Steps to Reproduce**:

Compile the following code fragment with **IDE0051** enabled:
```c#
internal static class Polyfill_Int32
{
    extension(int i)
    {
        public bool IsZero()
        {
            return IsZeroHelper(i);
        }

        private static bool IsZeroHelper(int value)
        {
            return value == 0;
        }
    }
}
```

**Expected Behavior**:

**IDE0051** is not reported.

**Actual Behavior**:

```txt
error IDE0051: Private member 'Polyfill_Int32.extension(Int32).IsZeroHelper' is unused
```

**Comment by CyrusNajmabadi** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/81754 and https://github.com/dotnet/roslyn/pull/81996
